index.js

```diff
@@ -1,12 +1,113 @@
 const express = require('express');
-const helmet = require('helmet');
+const configMiddleware = require('./config/middleware.js');
+
+const knex = require('knex');
+const knexConfig = require('./knexfile.js');
 
 const server = express();
 
-server.use(express.json());
-server.use(helmet());
+// apply middleware
+configMiddleware(server);
+
+// connect to database
+const db = knex(knexConfig.development);
+
+// generic 500 error handler
+const errorHandler = (err) => res.status(500).json({
+    message: 'Sorry, an error occurred while handling your request',
+    error: err
+  });
+
+// endpoints start here
+
+// list zoos
+server.get('/api/zoos', async (req, res) => {
+
+  try {
+    const zoos = await db('zoos');
+    res.status(200).json(zoos);
+  } catch (err) {
+    errorHandler(err);
+  }
+
+});
+
+// get zoo by ID
+server.get('/api/zoos/:id', async (req, res) => {
```

Comment:
Another option for routing with express is using the [`route`](https://expressjs.com/en/4x/api.html#app.route) method, like so:
```javascript
server.route('/api/zoos')
  .get(async (req, res) => { /* GET callback logic */ })
  .post(async (req, res) => { /* POST callback logic */ });

server.route('/api/zoos/:id')
  .get(async (req, res) => { /* GET callback logic */ })
  .put(async (req, res) => { /* PUT callback logic */ })
  .delete(async (req, res) => { /* DELETE callback logic */ });
```

Reply:
When people use this way of doing it, they often go ahead and make another file for the callback functions and store all them there. That way this will look short and clean (like it does now without a ton of database request logic packed in).